polkadot/xcm/xcm-builder/src/location_conversion.rs

```diff
@@ -328,6 +343,25 @@ impl<Network: Get<Option<NetworkId>>, AccountId: From<[u8; 32]> + Into<[u8; 32]>
 	}
 }
 
+/// Returns specified `TreasuryAccount` as `AccountId32` if passed `location` matches Treasury plurality.
+pub struct LocalTreasuryVoiceConvertsVia<TreasuryAccount, AccountId>(
+	PhantomData<(TreasuryAccount, AccountId)>,
+);
+impl<TreasuryAccount: Get<AccountId>, AccountId: From<[u8; 32]> + Into<[u8; 32]> + Clone>
+	ConvertLocation<AccountId> for LocalTreasuryVoiceConvertsVia<TreasuryAccount, AccountId>
+{
+	fn convert_location(location: &MultiLocation) -> Option<AccountId> {
+		let id: [u8; 32] = match *location {
```

Comment:
```suggestion
        match *location {
			MultiLocation {
				parents: 0,
				interior: X1(Plurality { id: BodyId::Treasury, part: BodyPart::Voice }),
			} => Some(TreasuryAccount::get().into()),
			_ => None,
		}
```

I think it should  be possible without temp variable